assets/js/phoenix.js

```diff
@@ -893,7 +894,7 @@ export class Socket {
     this.flushSendBuffer()
     this.reconnectTimer.reset()
     this.resetHeartbeat()
-    this.stateChangeCallbacks.open.forEach( callback => callback() )
+    this.stateChangeCallbacks.open.forEach(entry => entry[Object.keys(entry)[0]]() )
```

Comment:
ugh this doesn't look friendly.  We could go with a different data structure.  Right now this is an array of objects.

We could go with a "tuple" `open = [[topic, callback], [topic, callback]]`

Reply:
the tuple works, or we can do an object: `{ref, callback}`, your call